### DISCUSSION TITLE
Not working rules

### DISCUSSION THREAD

**Discussion by azu** [textlint/editor#31] [General]
We have collected not working rules on browser

- [textlint-rule-prh](https://github.com/textlint-rule/textlint-rule-prh)
  - dynamic fs.read
  - Probably, it will be difficult to resolve.
- [textlint-rule-spelling](https://github.com/tani/textlint-rule-spelling)
  - dynamic require + fs.readFile
- [textlint-rule-terminology](https://github.com/sapegin/textlint-rule-terminology)
  - fs.readFile
- ..[if you found, please add it]

## Need to idea

How to work these rules?

- Bundle a dictionary(Inlining)
- Dynamic loading a dictionary from URL

We need to provide helper library to resolve this issue?

**Comment by hata6502** [comment]
- textlint-rule-ja-hiragana-fukushi
  - fs.readFile
- textlint-rule-ja-hiragana-daimeishi
  - fs.readFile
- textlint-rule-ja-hiragana-hojodoushi
  - fs.readFile

BrowserFS may be able to resolve fs on browser environments. 
But I think it is not the best way because of the large bundle size. 

https://github.com/jvilk/BrowserFS
https://github.com/blue-hood/kohsei-san/commit/ff5192691260ab186a1e1dc7a4465166aeeb299a
https://github.com/blue-hood/kohsei-san/commit/a22654c2f15407f4e044e32c37b2cbbcf12f8f6d

**Comment by azu** [comment]
https://github.com/proofdict/proofdict/releases/tag/v3.1.1 work on textlint editor, now.
https://azu.github.io/textlint-editor-example/

**Comment by james-pre** [comment]
Hi @azu and @hata6502, I'm the maintainer of BrowserFS. I'd like to let you know that BrowserFS has been deprecated. You can read more about that here: https://github.com/browser-fs/NOTICE. [ZenFS](https://github.com/zen-fs/core) is an updated and active fork that I created.
